aries_cloudagent/tails/indy_tails_server.py

```diff
@@ -30,9 +32,21 @@ async def upload_tails_file(
             backoff: exponential backoff in retry interval
             max_attempts: maximum number of attempts to make
         """
-
-        genesis_transactions = context.settings.get("ledger.genesis_transactions")
         tails_server_upload_url = context.settings.get("tails_server_upload_url")
+        if not (context.settings.get("ledger.genesis_transactions")):
+            if isinstance(context, Profile):
+                context = context.context
+            pool = (
+                await context.injector.inject(
+                    BaseMultipleLedgerManager
+                ).get_write_ledger()
+            )[1].pool
+            try:
+                genesis_transactions = pool.genesis_transactions
+            except AttributeError:
+                genesis_transactions = pool.genesis_txns_cache
+        else:
+            genesis_transactions = context.settings.get("ledger.genesis_transactions")
```

Comment:
I think we could simplify this a bit and split up getting the pool into multiple lines for readability:

```suggestion
        genesis_transactions = context.settings.get("ledger.genesis_transactions")
        
        if not genesis_transactions:
            ledger_manager = context.injector.inject(BaseMultipleLedgerManager)    
            write_ledgers = await ledger.manager.get_write_ledger()
            pool = write_ledgers[0].pool
        
            try:
                genesis_transactions = pool.genesis_transactions
            except AttributeError:
                genesis_transactions = pool.genesis_txns_cache
```

Reply:
Updated accordingly